cloudsync/tests/test_provider.py

```diff
@@ -486,8 +486,7 @@ def pytest_generate_tests(metafunc):
             provs += [kw]
 
         if not provs:
-            provs += ["mock_oid_cs"]
```

Comment:
i'd rather use just the cs provs here, unless a test has some potential to cover case issues.   barring that this doesn't significantly test other things.

Reply:
I took this route because this seemed like an oversight, but it makes sense to not duplicate tests if we can avoid it.

Out of curiosity though -- why do you think we should only test case-sensitive mock providers, given that all the providers we care about at the moment are case-insensitive? Why not keep 2, and test one of each, say path_cs and oid_ci?

Reply:
I don't think we should only test case-sensitive mock providers.   I think we should test both.   Case sensitive providers are, in general, fairly common and there's no reason to care one way or another which one we use... for tests upon which case has no impact.

Reply:
So before this change, the default set of mocks was: mock_path_cs, mock_oid_cs (though, there was bug in mock_oid_cs where oid_is_path was set to True, so we were really testing mock_path_cs twice -- this is now fixed).

With my change as is we are now testing 4 combinations, which I agree is overkill. I think we could cut this back down to 2 if the default set includes a cs and a ci mock -- this is the simple solution in my opinion.

We certainly could parameterize case-sensitive tests such that they run through a different set of mocks than case-insensitive test, but that seems overly complicated. Path handling is a low level thing that is difficult to isolate.

Reply:
It's important to test both cs and ci for oid_is_path providers because oid_is_path providers are much more sensitive to problems with path handling than oid_is_oid providers

Reply:
I think we /must/ parameterize "case tests" different than other tests.

- Many tests are slow and running with multiple mocks isn't helpful
- oid/path vs oid/oid providers have different considerations regarding case
- 90% of our tests cannot possibly have case interaction
- Running with multiple mocks gets us "coverage" in case code... but it's not meaningful coverage, it masks the fact that we are not testing case.
- We must test things like case sensitive sync with case insensitive providers and have intelligent conflict handling.

I don't want to treat case like we can just turn it on and ignore it.

Reply:
I get it now... you want cs_oid, ci_path .... that will cover everything without duplicating stuff.